clients/stellar-relay/src/connection/connector/connector.rs

```diff
@@ -193,3 +193,218 @@ impl Connector {
+		assert_eq!(connector.receive_scp_messages(), connConfig.recv_scp_messages);
+
+		connector.got_hello();
+		assert!(connector.is_handshake_created());
+
+		connector.handshake_completed();
+		assert!(connector.is_handshake_created());
+	}
+
+	#[tokio::test]
+	async fn connector_send_to_user_works() {
+		let (_, _, mut connector, _, mut message_receiver) = create_connector();
+
+		let message = StellarRelayMessage::Timeout;
+		connector.send_to_user(message).await.unwrap();
+
+		let received_message = message_receiver.recv().await;
+		assert!(received_message.is_some());
+		let message = received_message.unwrap();
+		match message {
+			StellarRelayMessage::Timeout => {},
+			_ => {
+				panic!("Incorrect message received!!!")
+			},
+		}
+	}
+
+	#[test]
+	fn enable_flow_controller_works() {
+		let (node_info, _, mut connector, _, _) = create_connector();
+
+		assert!(!connector.inner_check_to_send_more(MessageType::ScpMessage));
+		connector.enable_flow_controller(node_info.overlay_version, node_info.overlay_version);
+	}
+
+	#[tokio::test]
+	async fn connector_send_to_node_works() {
+		let (_, _, mut connector, mut actions_receiver, _) = create_connector();
+
+		connector.send_to_node(ConnectorActions::SendHello).await.unwrap();
+
+		let received_message = actions_receiver.recv().await;
+		assert!(received_message.is_some());
+		let message = received_message.unwrap();
+		match message {
+			ConnectorActions::SendHello => {},
+			_ => {
+				panic!("Incorrect message received!!!")
+			},
+		}
+	}
+
+	fn create_auth_cert_from_connection_auth(
+		connector_auth: &ConnectionAuth,
+	) -> substrate_stellar_sdk::types::AuthCert {
+		let time_now = time_now();
+		let new_auth_cert = create_auth_cert(
+			connector_auth.network_id(),
+			connector_auth.keypair(),
+			time_now,
+			connector_auth.pub_key_ecdh().clone(),
+		)
+		.expect("should successfully create an auth cert");
+		new_auth_cert
+	}
+
+	fn create_connector(
+	) -> (NodeInfo, ConnConfig, Connector, Receiver<ConnectorActions>, Receiver<StellarRelayMessage>)
+	{
+		use substrate_stellar_sdk::{network::TEST_NETWORK, SecretKey};
+		let secret =
+			SecretKey::from_encoding("SBLI7RKEJAEFGLZUBSCOFJHQBPFYIIPLBCKN7WVCWT4NEG2UJEW33N73")
+				.unwrap();
+		let node_info = NodeInfo::new(19, 21, 19, "v19.1.0".to_string(), &TEST_NETWORK);
+		let cfg = ConnConfig::new("34.235.168.98", 11625, secret, 0, false, true, false);
+		// this is a channel to communicate with the connection/config (this needs renaming)
+		let (actions_sender, actions_receiver) = mpsc::channel::<ConnectorActions>(1024);
+		// this is a channel to communicate with the user/caller.
+		let (relay_message_sender, relay_message_receiver) =
+			mpsc::channel::<StellarRelayMessage>(1024);
+		let connector =
+			Connector::new(node_info.clone(), cfg.clone(), actions_sender, relay_message_sender);
+		(node_info, cfg, connector, actions_receiver, relay_message_receiver)
+	}
```

Comment:
I would move these two functions to the top of the test module so they are known to the reader of the code before being used.